test/SlowTests/Issues/RavenDB-17451.cs

```diff
@@ -0,0 +1,43 @@
+using System.Threading.Tasks;
+using FastTests;
+using SlowTests.Core.AdminConsole;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_17451 : RavenTestBase
+    {
+        public RavenDB_17451(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task CanModifyDisableDatabaseTcpCompressionConfiguration()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var database = await GetDocumentDatabaseInstanceFor(store);
+                var configuration = database.Configuration;
+
+                Assert.False(configuration.Databases.DisableTcpCompression);
+
+                AdminJsConsoleTests.ExecuteScript(Server, database, "database.Configuration.Databases.DisableTcpCompression = true;");
```

Comment:
This actually tests that the JS console works correctly, not that the configuration value was applied correctly according to user's setup. You should rather create a new db with relevant setting in the db record in order to check that. However what would be more important I think is to check that the client (which has `Convetions.DisableTcpCompression: false`) can still talk to a database without any issue.

Reply:
Given that this is going to be a server wide option, we need to setup a new server entirely.

Reply:
fixed, added more tests